incubator/blockchain-client-openliberty/templates/default/src/main/java/application/api/AssetService.java

```diff
@@ -76,6 +76,10 @@ public Response getMyAsset(@PathParam("id") String id) throws AssetNotFoundExcep
     @Tag(name = "MyAssets")
     public Response updateMyAsset(@PathParam("id") String id, MyAsset asset) throws IdentityException, GatewayException {
         LOGGER.info("Asset : " + asset.toString());
+        // Give preference to the value passed in the path rather than a possible value in the payload. 
+        if(!asset.getMyAssetId().equals(id)){
+            asset.setMyAssetId(id);
```

Comment:
I am wondering if we should throw an exception if the two IDs do not match...

Reply:
Great suggestion.